pkgs/applications/misc/1password/default.nix

```diff
@@ -26,13 +26,22 @@ stdenv.mkDerivation rec {
     zcat Payload | cpio -i
   '';
 
+  dontStrip = stdenv.isDarwin;
+
+  postPhases = [ "postInstall" ];
```

Comment:
I am not sure if it is cool overwriting this phase.

Reply:
Without overwriting this, the phase never runs and shell completion never gets installed. I'm not sure there's a better solution.

Reply:
That is because installPhase is overwritten and not calling pre and post hooks.

Reply:
When you do call those hooks, the binary never gets patched. Not sure how to approach this, diff i used + error are below.
```diff
diff --git a/pkgs/applications/misc/1password/default.nix b/pkgs/applications/misc/1password/default.nix
index 9e52e1ac244..56d8d402e15 100644
--- a/pkgs/applications/misc/1password/default.nix
+++ b/pkgs/applications/misc/1password/default.nix
@@ -28,12 +28,12 @@ stdenv.mkDerivation rec {

   dontStrip = stdenv.isDarwin;

-  postPhases = [ "postInstall" ];
-
   nativeBuildInputs = [ installShellFiles ] ++ stdenv.lib.optionals stdenv.isLinux [ autoPatchelfHook ];

   installPhase = ''
+    runHook preInstall
     install -D op $out/bin/op
+    runHook postInstall
   '';

   # This needs to be done after patchelf has been ran on the binary
```
```shell
building '/nix/store/pb0c6xx470ir0bij7b1qyv99f3g2fa5h-1password-1.8.0.drv'...
builder for '/nix/store/pb0c6xx470ir0bij7b1qyv99f3g2fa5h-1password-1.8.0.drv' failed with exit code 127; last 10 log lines:
  unpacking sources
  unpacking source archive /nix/store/cnlnslraf8zlw1c2158bdl7nnkw94gdg-source
  source root is source
  patching sources
  configuring
  no configure script, doing nothing
  building
  no Makefile, doing nothing
  installing
  /nix/store/7d2w3lafm8cla7n6ii5jskhsnqg4zv3a-stdenv-linux/setup: line 81: /nix/store/n0j5i8a7x1y3clibjvyfs0xb5ll2snf5-1password-1.8.0/bin/op: No such file or directory
[0 built (1 failed)]
error: build of '/nix/store/pb0c6xx470ir0bij7b1qyv99f3g2fa5h-1password-1.8.0.drv' failed
~/nix/nixpkgs/pkgs/applications/misc/1password > file /nix/store/n0j5i8a7x1y3clibjvyfs0xb5ll2snf5-1password-1.8.0/bin/op
/nix/store/n0j5i8a7x1y3clibjvyfs0xb5ll2snf5-1password-1.8.0/bin/op: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, Go BuildID=JIw--AYYM5GRNoEakTfx/NT4DSbHHI2jWoJQdw-mg/Wp_rILnTzXvcSq99zeAp/9Dl6huS2F26iSE1Jm78q, stripped

```